mpmath/ctx_mp_python.py

```diff
@@ -228,8 +228,50 @@ def ae(s, t, rel_eps=None, abs_eps=None):
     def to_fixed(self, prec):
         return to_fixed(self._mpf_, prec)
 
-    def __round__(self, *args):
-        return round(float(self), *args)
+    def __round__(self, n=0):
+        """
+        Round to `n` digits, respecting current rounding flags and precision.
+
+        The Python docs [1] say:
+        > For the built-in types ... toward the even choice
+
+        Thus, as we are not a built-in type, we can round using the current
+        rounding flag.
+
+        This implementation always returns an `mpf` object.
+        TODO: would we like `round(foo, 0)` to return `int`?
```

Comment:
"The return value is an integer if ndigits is omitted or None. Otherwise, the return value has the same type as number." (c) CPython
But I think we also can ignore this:)